test/Microsoft.Framework.DesignTimeHost.Tests/PluginHandlerFacts.cs

```diff
@@ -0,0 +1,491 @@
+            var pluginHandler = new PluginHandler(serviceProvider, (_) => { });
+            var pluginMessage = new PluginMessage
+            {
+                Data = new JObject
+                {
+                    { "AssemblyName", "CustomAssembly" },
+                    { "TypeName", pluginType.FullName },
+                },
+                MessageName = "RegisterPlugin",
+                PluginId = RandomGuidId
+            };
+
+            pluginHandler.ProcessMessage(pluginMessage, assemblyLoadContext);
+
+            Assert.False(creationChecker.Created);
+        }
+
+        [Fact]
+        public void ProcessMessage_RegisterPlugin_ThrowsOnInvalidPluginTypes()
+        {
+            var assemblyLoadContext = CreateTestAssemblyLoadContext<InvalidTestPlugin>();
+            var serviceProvider = new TestServiceProvider();
+            var pluginHandler = new PluginHandler(serviceProvider, (_) => { });
+            var pluginMessage = new PluginMessage
+            {
+                Data = new JObject
+                {
+                    { "AssemblyName", "CustomAssembly" },
+                    { "TypeName", typeof(InvalidTestPlugin).FullName },
+                },
+                MessageName = "RegisterPlugin",
+                PluginId = RandomGuidId
+            };
+            var expectedErrorMessage =
+                $"Cannot process plugin message. Plugin id '{RandomGuidId}' of type " +
+                "'Microsoft.Framework.DesignTimeHost.PluginHandlerFacts+InvalidTestPlugin' must be assignable " +
+                "to type 'Microsoft.Framework.DesignTimeHost.IPlugin'.";
+
+            var error = Assert.Throws<InvalidOperationException>(
+                () => pluginHandler.ProcessMessage(pluginMessage, assemblyLoadContext));
+            Assert.Equal(expectedErrorMessage, error.Message, StringComparer.Ordinal);
+        }
+
+        [Fact]
+        public void ProcessMessage_UnregisterPlugin_UnregistersPlugin()
+        {
+            var assemblyLoadContext = CreateTestAssemblyLoadContext<TestPlugin>();
+            var serviceProvider = new TestServiceProvider();
+            var pluginHandler = new PluginHandler(serviceProvider, (_) => { });
+            var registerPluginMessage = new PluginMessage
+            {
+                Data = new JObject
+                {
+                    { "AssemblyName", "CustomAssembly" },
+                    { "TypeName", typeof(TestPlugin).FullName },
+                },
+                MessageName = "RegisterPlugin",
+                PluginId = RandomGuidId
+            };
+            var unregisterPluginMessage = new PluginMessage
+            {
+                MessageName = "UnregisterPlugin",
+                PluginId = RandomGuidId
+            };
+
+            pluginHandler.ProcessMessage(registerPluginMessage, assemblyLoadContext);
+            pluginHandler.ProcessMessage(unregisterPluginMessage, assemblyLoadContext);
```

Comment:
test shows only that these two `ProcessMessage()` calls execute without throwing.  need to confirm something deeper.